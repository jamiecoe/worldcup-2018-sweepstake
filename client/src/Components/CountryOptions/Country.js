import React from 'react';
import styled, { css } from 'styled-components';

const StyledDiv = styled.div`
  margin: 10px;
  text-align: center;
  position: relative;  
  min-width: 240px;
  min-height: 125px;

  ${props =>
    props.knockedOut &&
    css`
      &:before {
        content: '';
        display: block;
        position: absolute;
        top: 0;
        bottom: 0;
        left: 0;
        right: 0;
        background: rgba(0, 0, 0, 0.5);
      }
    `};
`;

const OverlayDiv = styled.div`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  height: 100%;
  width: 100%;
  opacity: 0;
  transition: 0.4s ease;
  background-color: rgba(0, 0, 0, 0.6);
  display: flex;
  justify-content: center;
  align-items: center;

  &:hover {
    opacity: 0.9;
  }
`;

const StyledImg = styled.img`
  box-shadow: 0 0 4px 1px rgba(0, 0, 0, 0.3);
  height: 100%;  
  width: 100%;
`;

const Country = props => {
  return (
    <StyledDiv knockedOut={props.knockedOut}>
      <StyledImg src={props.imgUrl} alt={`${props.name} flag`} />
      <OverlayDiv>
        <h3>{props.name}</h3>
      </OverlayDiv>
    </StyledDiv>
  );
};

export default Country;
