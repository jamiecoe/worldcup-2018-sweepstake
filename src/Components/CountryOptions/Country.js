import React from 'react';

const Country = (props) => {
  console.log(props);
  
  return (
    <div>
      <img src={props.imgUrl} alt={`${props.name} flag`}/>
      <h3>{props.name}</h3>
    </div>
  )
}

export default Country;
