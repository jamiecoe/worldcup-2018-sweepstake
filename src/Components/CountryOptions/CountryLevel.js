import React, { Component } from 'react';

class CountryLevel extends Component {
  render() {
    return (
      <div>
        <h2>{this.props.level}</h2>
      </div>
    )
  }
}

export default CountryLevel;
