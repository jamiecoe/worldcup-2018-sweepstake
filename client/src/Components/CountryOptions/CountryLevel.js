import React, { Component } from 'react';
import styled from 'styled-components';
import Country from './Country';

const StyledDiv = styled.div`
  display: flex;
  overflow-x: scroll;
  box-shadow: 0 0 4px 1px rgba(0, 0, 0, 0.3);
  background: white;
`

class CountryLevel extends Component {
  displayCountries = () => {
    const { countries } = this.props;  
    const componentArray = [];

    for (const country in countries) {
      componentArray.push(<Country {...countries[country]} key={countries[country].name} />);
    }

    return componentArray;
  }
    

  render() {
    return (
      <div>
        <h2>{this.props.level}</h2>
        <StyledDiv>          
          {this.displayCountries()}
        </StyledDiv>
      </div>
    );
  }
}

export default CountryLevel;
