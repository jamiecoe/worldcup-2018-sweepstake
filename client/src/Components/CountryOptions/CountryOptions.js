import React from 'react';
import styled from 'styled-components';
import CountryLevel from './CountryLevel';

const StyledDiv = styled.div`
  width: 100%;
`

const CountryOptions = (props) => {

    const { top, mid, low } = props.countries;

    return (
      <StyledDiv>
        <h2>Available teams :</h2>
        <CountryLevel countries={top} level="Top" />
        <CountryLevel countries={mid} level="Mid" />
        <CountryLevel countries={low} level="Low" />
      </StyledDiv>
    );
}

export default CountryOptions;
