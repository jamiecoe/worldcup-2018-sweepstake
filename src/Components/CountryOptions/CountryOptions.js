import React from 'react';
import CountryLevel from './CountryLevel';

const CountryOptions = (props) => {

    const { top, mid, low } = props.countries;

    return (
      <div>
        <CountryLevel countries={top} level="Top" />
        <CountryLevel countries={mid} level="Mid" />
        <CountryLevel countries={low} level="Low" />
      </div>
    );
}

export default CountryOptions;
