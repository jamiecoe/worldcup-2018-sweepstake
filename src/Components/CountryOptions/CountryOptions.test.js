import React from 'react';
import { shallow } from 'enzyme';
import CountryOptions from './CountryOptions';

describe('CountryOptions', () => {
  const props = {
    countries: {
      top: {},
      mid: {},
      low: {},
    },
  };
  const countryOptions = shallow(<CountryOptions {...props} />);

  it('renders properly', () => {
    expect(countryOptions).toMatchSnapshot();
  });
});
