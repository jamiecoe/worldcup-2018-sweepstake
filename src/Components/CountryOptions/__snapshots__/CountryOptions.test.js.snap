// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryOptions renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CountryOptions
    countries={
        Object {
            "low": Object {},
            "mid": Object {},
            "top": Object {},
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <h2>
          Available teams :
</h2>,
        <CountryLevel
          countries={Object {}}
          level="Top"
/>,
        <CountryLevel
          countries={Object {}}
          level="Mid"
/>,
        <CountryLevel
          countries={Object {}}
          level="Low"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Available teams :",
        },
        "ref": null,
        "rendered": "Available teams :",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "countries": Object {},
          "level": "Top",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "countries": Object {},
          "level": "Mid",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "countries": Object {},
          "level": "Low",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <h2>
            Available teams :
</h2>,
          <CountryLevel
            countries={Object {}}
            level="Top"
/>,
          <CountryLevel
            countries={Object {}}
            level="Mid"
/>,
          <CountryLevel
            countries={Object {}}
            level="Low"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Available teams :",
          },
          "ref": null,
          "rendered": "Available teams :",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "countries": Object {},
            "level": "Top",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "countries": Object {},
            "level": "Mid",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "countries": Object {},
            "level": "Low",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
